gensim/similarities/nmslib.py

```diff
@@ -0,0 +1,210 @@
+            >>> model = Word2Vec(sentences, min_count=1, seed=1)
+            >>>
+            >>> indexer = NmslibIndexer(model)
+            >>> model.most_similar("cat", topn=2, indexer=indexer)
+            [('cat', 1.0), ('meow', 0.5595494508743286)]
+
+        """
+        if index_params is None:
+            index_params = {'M': 10, 'indexThreadQty': 1, 'efConstruction': 100, 'post': 0}
+        if query_time_params is None:
+            query_time_params = {'efSearch': 100}
+
+        self.index = None
+        self.labels = None
+        self.model = model
+        self.index_params = index_params
+        self.query_time_params = query_time_params
+
+        if model:
+            if isinstance(self.model, Doc2Vec):
+                self.build_from_doc2vec()
+            elif isinstance(self.model, (Word2Vec, FastText)):
+                self.build_from_word2vec()
+            elif isinstance(self.model, (WordEmbeddingsKeyedVectors, KeyedVectors)):
+                self.build_from_keyedvectors()
+            else:
+                raise ValueError("Only a Word2Vec, Doc2Vec, FastText or KeyedVectors instance can be used")
+
+    def save(self, fname, protocol=2):
+        """Save NmslibIndexer instance.
+
+        Parameters
+        ----------
+        fname : str
+            Path to output file,
+            will produce 2 files: `fname` - parameters and `fname`.d - :class:`~nmslib.NmslibIndex`.
+        protocol : int, optional
+            Protocol for pickle.
+
+        Notes
+        -----
+        This method save **only** index (**model isn't preserved**).
+
+        """
+        fname_dict = fname + '.d'
+        self.index.saveIndex(fname)
+        d = {'index_params': self.index_params, 'query_time_params': self.query_time_params, 'labels': self.labels}
+        with open(fname_dict, 'wb') as fout:
+            _pickle.dump(d, fout, protocol=protocol)
+
+    def load(self, fname):
+        """Load NmslibIndexer instance
+
+        Parameters
+        ----------
+        fname : str
+            Path to dump with NmslibIndexer.
+
+        Examples
+        --------
+        .. sourcecode:: pycon
+
+            >>> from gensim.similarities.nmslib import NmslibIndexer
+            >>> from gensim.models import Word2Vec
+            >>> from tempfile import mkstemp
+            >>>
+            >>> sentences = [['cute', 'cat', 'say', 'meow'], ['cute', 'dog', 'say', 'woof']]
+            >>> model = Word2Vec(sentences, min_count=1, seed=1, iter=10)
+            >>>
+            >>> indexer = NmslibIndexer(model)
+            >>> _, temp_fn = mkstemp()
+            >>> indexer.save(temp_fn)
+            >>>
+            >>> new_indexer = NmslibIndexer()
+            >>> new_indexer.load(temp_fn)
+            >>> new_indexer.model = model
+
+        """
+        fname_dict = fname + '.d'
+        with open(fname_dict, 'rb') as f:
+            d = _pickle.loads(f.read())
```

Comment:
Doesn't pickle support reading directly from files?

```suggestion
            d = _pickle.load(f)
```

Reply:
Thanks. Changed to use pickle.load instead of pickle.loads.